wfl/fit/mace.py

```diff
@@ -0,0 +1,205 @@
+    run_dir = Path(run_dir)
+    
+    # If fitting_configs is not given as a file but a memory, 
+	# It should be first written as xyz file. 
+    if not fitting_configs.one_file():
+        fitting_configs = prepare_configs(fitting_configs)
+        # Not so certain how mace_file_base should be set
+        # In iterative training, it should be iteration index.
+        mace_file_base = "test"
+        _write_fitting_configs(fitting_configs, mace_fit_params, mace_file_base)
+
+
+    if skip_if_present:
+        try:
+            print(f"check whether already fitted model exists as {run_dir}/{mace_name}.model")
+            if not Path(f"{run_dir}/{mace_name}.model").is_file():
+                raise FileNotFoundError
+    
+            return mace_name
+        except (FileNotFoundError, RuntimeError):
+            pass
+    
+    
+    if remote_info != '_IGNORE':
+        remote_info = get_remote_info(remote_info, remote_label)
+    
+    
+    if remote_info is not None and remote_info != '_IGNORE':
+        input_files = remote_info.input_files.copy()
+        output_files = remote_info.output_files.copy() + [str(run_dir)]
+    
+    
+        # set number of threads in queued job, only if user hasn't set them
+        if not any([var.split('=')[0] == 'WFL_GAP_FIT_OMP_NUM_THREADS' for var in remote_info.env_vars]):
+            remote_info.env_vars.append('WFL_GAP_FIT_OMP_NUM_THREADS=$EXPYRE_NUM_CORES_PER_NODE')
+        if not any([var.split('=')[0] == 'WFL_NUM_PYTHON_SUBPROCESSES' for var in remote_info.env_vars]):
+            remote_info.env_vars.append('WFL_NUM_PYTHON_SUBPROCESSES=$EXPYRE_NUM_CORES_PER_NODE')
+    
+        remote_func_kwargs = {'mace_fit_params': mace_fit_params,'remote_info': '_IGNORE', 'run_dir': run_dir,
+                            'input_files' : remote_info.input_files.copy(), 
+                            "mace_fit_cmd" : mace_fit_cmd}
+    
+        kwargs.update(remote_func_kwargs)
+        xpr = ExPyRe(name=remote_info.job_name, pre_run_commands=remote_info.pre_cmds, post_run_commands=remote_info.post_cmds,
+                     env_vars=remote_info.env_vars, input_files=input_files, output_files=output_files, function=fit,
+                     kwargs = remote_func_kwargs)
+    
+        xpr.start(resources=remote_info.resources, system_name=remote_info.sys_name, header_extra=remote_info.header_extra,
+                  exact_fit=remote_info.exact_fit, partial_node=remote_info.partial_node)
+    
+        if not wait_for_results:
+            return None
+        results, stdout, stderr = xpr.get_results(timeout=remote_info.timeout, check_interval=remote_info.check_interval)
+    
+        sys.stdout.write(stdout)
+        sys.stderr.write(stderr)
+    
+        # no outputs to rename since everything should be in run_dir
+        xpr.mark_processed()
+        
+        return results
+    
+    if not run_dir.exists():
+        run_dir.mkdir(parents=True)
+    
+    if isinstance(mace_fit_params, str):
```

Comment:
I'd prefer to simplify this function by requiring that the params be passed to this function in the form of a dict.  If the calling function wants to load that dict from a yaml, that's fine, but isn't `fit()`'s business.